### DISCUSSION TITLE
Vacuum Service - clean a zone

### DISCUSSION THREAD

**Discussion by fabicro** [Tasshack/dreame-vacuum#40] [Q&A] [answered]
I would like to generate a script to be able to clean a specific zone, as explained in this video.
[https://www.youtube.com/watch?v=-UXL4a_fmOc](url)
Unfortunately if I try to call the service as indicated

service: dreame_vacuum.vacuum_clean_zone`
data: 
   zone: 
     - 819
     - -263
     - 4424
     - 2105
target:
   entity_id: vacuum.vacuum

i get this error

None @ data['zone'][0]

instead if I do it through the Xiaomi Vacuum Card it works perfectly.

[https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card](url)

I don't understand if and how to specify the map where the points of the selected zone come from.

**Comment by Tasshack** [comment, marked as answer]
App only allows you to select single area but device is capable of cleaning multiple zones on a single task, because of that you need to pass zone coordinates array inside of another array when sending the action to the device.

```
service: dreame_vacuum.vacuum_clean_zone
data:
  zone:
    - - 819
      - -263
      - 4424
      - 2105
target:
  entity_id: vacuum.vacuum
```

Actually integration should handle the single zone with this service but the validation is not allows passing a single array right now, i will fix this on next release.

> I don't understand if and how to specify the map where the points of the selected zone come from.

[https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card/blob/master/docs/templates/setup.md#getting-coordinates](https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card/blob/master/docs/templates/setup.md#getting-coordinates)

**Comment by fabicro** [comment]
Clear. So I wait for the new release. In the meantime, thank you for your very useful support.
